operator/internal/manifests/compactor.go

```diff
@@ -26,7 +26,7 @@ func BuildCompactor(opts Options) ([]client.Object, error) {
 		}
 	}
 
-	if err := storage.ConfigureStatefulSet(statefulSet, opts.ObjectStorage); err != nil {
+	if err := storage.ConfigureStatefulSet(statefulSet, opts.ObjectStorage, opts.Gates.OpenShift.Enabled); err != nil {
```

Comment:
When we need more than one field from `Options`, I suggest to pass the whole object instead of exploding it in parameters. This encapsulates usage in the callee site and in turn when you look up for references of Options fields you have getters/setters always within the context of the callee site.

Reply:
I'm not able to import the `manifests.Options` as that creates a circular dependency. After some consideration, I believe we have 2 options:
1. We set the SATokeFilePath before reaching the manifests generation (I was thinking after extract secret)
2. We replicate the OpenShift enabled fields inside `storage.Options` (This was what i ended up following, let me know how it looks)